index.js

```diff
@@ -65,25 +70,35 @@ function fastifyStatic (fastify, opts, next) {
   next()
 }
 
-function checkRootPathForErrors (rootPath) {
-  if (typeof rootPath !== 'string') {
-    return new Error('"root" option is required')
-  }
+function checkPathsForErrors (paths) {
+  if (paths.root === undefined) return new Error('"root" option is required')
 
-  if (path.isAbsolute(rootPath) === false) {
-    return new Error('"root" option must be an absolute path')
-  }
+  for (const p in paths) {
+    if (paths[p] !== undefined) {
+      if (typeof paths[p] !== 'string') {
+        return new Error(`"${p}" option must be a string`)
+      }
 
-  let rootStat
+      if (path.isAbsolute(paths[p]) === false) {
+        return new Error(`"${p}" option must be an absolute path`)
+      }
 
-  try {
-    rootStat = statSync(rootPath)
-  } catch (e) {
-    return e
-  }
+      let pathStat
 
-  if (rootStat.isDirectory() === false) {
-    return new Error('"root" option must point to a directory')
+      try {
+        pathStat = statSync(paths[p])
+      } catch (e) {
+        return e
+      }
+
+      if (p === 'root' && pathStat.isDirectory() === false) {
```

Comment:
I don't like this "if" inside a for loop.

I'd like more something like
```js
function checkPathsForErrors (paths) {
  checkTypePath(paths.root, 'isDirectory')
  if (paths.page500Path) checkTypePath(paths.page500Path, 'isFile')
  if (paths.page403Path) checkTypePath(paths.page403Path, 'isFile')
  if (paths.page404Path) checkTypePath(paths.page404Path, 'isFile')
}
```
where `checkTypePath` is a function that checks the path existance and the type

**NB:** you can use the second string to access the `pathStat` method and call it checking if it returns `true`

Reply:
Good suggestions — refactored!